src/app/components/footer/footer.component.ts

```diff
@@ -1,9 +1,19 @@
 import {Component} from '@angular/core'
+import {HeaderComponent} from '../header/header.component'
 
 @Component({
+    providers: [HeaderComponent],
     selector: 'app-footer',
     templateUrl: './footer.component.html',
     styleUrls: ['./footer.component.scss']
 })
 export class FooterComponent {
+    constructor(private headerComponent: HeaderComponent) {
```

Comment:
injecting components like this is an `anti-pattern`. The correct way of fixing this is to have a `nightMode` service and move the logic there. Then components can inject `NightModeService` and use that to check for night mode.

Reply:
thx for pointing that out to me!